### PR TITLE
4-6% speedup in "angular_separation" by using hypot ufunc

### PR DESCRIPTION
**Pull request by scienceopen**
4-6% speedup observed in angular_separation computation by using ```numpy.hypot(x,y)``` instead of ```numpy.sqrt(x**2 + y**2)```

### CODE REVIEW

File: astropy/coordinates/angle_utilities.py [hunk 1]
```diff
@@ -655,7 +655,7 @@ def angular_separation(lon1, lat1, lon2, lat2):
     num2 = clat1 * slat2 - slat1 * clat2 * cdlon
     denominator = slat1 * slat2 + clat1 * clat2 * cdlon
 
-    return np.arctan2(np.sqrt(num1 ** 2 + num2 ** 2), denominator)
+    return np.arctan2(np.hypot(num1,num2), denominator)
```
Comment (mhvk) [inline astropy/coordinates/angle_utilities.py]: Looks good, but one nit-pick: please put a space after the ',' (PEP8)

**Comment by taldcroft** [comment]
Sorry, I can't resist on commenting that you can speed up `angular_separation` by 700 to 1700% by not using it and instead using a unit-free implementation: see `sphere_dist` in #3908.  :stuck_out_tongue_winking_eye:  But of course squeezing a bit out of the astropy `angular_separation` is a good thing.

**Comment by mhvk** [comment]
@taldcroft - taking the bait: this routine doesn't require units at all: you can just pass in arrays in radians; the main reason for the speed-up in your implementation in #3908 is that you use `numexpr`. With units, this routine should loose about a factor of 2 if input is not in radian, since the expensive trig calculations are done twice (this will become much better when `__numpy_ufunc__` finally gets implemented in `numpy`).

**Comment by taldcroft** [comment]
@mhvk - and it's true to be fair my implementation is Haversine vs. Vincenty (the latter is known to be slower).  But now I'm wondering why Vincenty is used here since it looks like the accuracy improvement is primarily related to its use on an oblate spheroid (i.e. the Earth).  Does anyone know if Vincenty is more accurate at the poles than Haversine for a perfect spheroid?

Regarding use of numexpr, this is an increasingly easy dependency (default in Anaconda), so what about using an alternate code path if numexpr is available (and values are floats)?

**Comment by mhvk** [comment]
From previous discussions, I think we settled on just using the one with best over-all-the-sky accuracy (IIRC, haversine is bad for sources on nearly opposite sides of the sky). 

As for `numexpr`: this might well be a good idea, but seems to go quite far outside of the scope of this PR!

**Comment by mhvk** [comment]
@scienceopen - could you fix the extra space; I'll merge after.

**Comment by mhvk** [comment]
@scienceopen - I went ahead and made the PEP8 correction myself, in #4643, still keeping credit for the commit to you.  I'll merge that one. Thanks!

**Pull request closed without merging**